### PR TITLE
JIT: Remove "add copies" phase

### PR DESCRIPTION
**Pull request by jakobbotsch**
This phase seems to try to do some ad-hoc live range splitting to improve things in assertion prop, but it almost barely kicks in due to the following checks:

https://github.com/dotnet/runtime/blob/588dcd7a29f109daae2d8c999322acd8e853a8c6/src/coreclr/jit/assertionprop.cpp#L554-L580

When it does kick in it seems to overall be a regression ([diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=201947&view=ms.vss-build-web.run-extensions-tab)), both in ASM diffs and in TP diffs.
Furthermore, we pay 16 bytes in every `LclVarDsc` (out of 88) for bookkeeping purposes for this pass, even in MinOpts.

### CODE REVIEW

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -4827,14 +4827,6 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     // However, ref counts are not kept incrementally up to date.
     assert(lvaLocalVarRefCounted());
 
-    if (opts.OptimizationEnabled())
-    {
-        // Introduce copies for some single-def locals to make them more
-        // amenable to optimization
-        //
-        DoPhase(this, PHASE_OPTIMIZE_ADD_COPIES, &Compiler::optAddCopies);
```
Comment (EgorBo) [inline src/coreclr/jit/compiler.cpp:4835]: Remove `PHASE_OPTIMIZE_ADD_COPIES` from compphases.h ?

**Comment by EgorBo** [comment]
Nice wins actually 😄

**Comment by jakobbotsch** [comment]
> Nice wins actually 😄 (except TP on arm32)

This is likely just an artifact of the decreased size of `LclVarDsc`, I will push this without the fields removed to see

**Comment by jakobbotsch** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=201947&view=ms.vss-build-web.run-extensions-tab) when the fields are left in, but with the pass itself removed.

cc @dotnet/jit-contrib -- I propose that we remove this phase. The phase itself is overall a regression to have around, both on TP and ASM diffs. In addition it requires 16 bytes in `LclVarDsc` (out of 88 on x64) that we pay for even in MinOpts.
The phase seems to try to do some ad-hoc live-range splitting before assertion prop, but it almost never kicks in because of these checks:

https://github.com/dotnet/runtime/blob/588dcd7a29f109daae2d8c999322acd8e853a8c6/src/coreclr/jit/assertionprop.cpp#L554-L580

The first check there also means it kicks in even less in customer code that don't have `SkipLocalsInit`.

File: src/coreclr/jit/compiler.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/compiler.cpp:4835]: Whoops, good catch, done!

**Review by EgorBo — APPROVED** [review]

**Pull request merged**